logstash-core/src/main/java/org/logstash/plugins/PluginFactoryExt.java

```diff
@@ -106,10 +121,18 @@ public Plugins(final Ruby runtime, final RubyClass metaClass) {
         public PluginFactoryExt.Plugins init(final PipelineIR lir, final PluginFactoryExt.Metrics metrics,
                                              final PluginFactoryExt.ExecutionContext executionContext,
                                              final RubyClass filterClass) {
+            return this.init(lir, metrics, executionContext, filterClass, EnvironmentVariableProvider.defaultProvider());
+        }
+
+        PluginFactoryExt.Plugins init(final PipelineIR lir, final PluginFactoryExt.Metrics metrics,
+                                      final PluginFactoryExt.ExecutionContext executionContext,
+                                      final RubyClass filterClass,
+                                      final EnvironmentVariableProvider envVars) {
             this.lir = lir;
             this.metrics = metrics;
             this.executionContext = executionContext;
             this.filterClass = filterClass;
+            this.configVariables = ConfigVariableExpander.withoutSecret(envVars);
```

Comment:
Right now the expectation we set in docs is that `${XXX}` are _always_ sourced from the secret store first, and only fall back to environment variable if there is no secret (see: [docs/static/keystore.asciidoc@master](https://github.com/elastic/logstash/blob/master/docs/static/keystore.asciidoc))

I can see how mapping the secret store in here would greatly increase the complexity, and see little value in supporting ids as secrets when they are now being included in logs etc., but if ids don't support secrets stored in the secret store, then we should note this in documentation.

Reply:
Thanks @yaauie  to have spotted this!
I thought that `id` has not meaning to grab variables from secret store because they are exposed through HTTP API and logged, so I intentionally left out the secret store substitution. I've added a note in the doc, let me know if it's the right place or we should document in other places/ways

Reply:
The note you added in the variable substitution doc is helpful, but I think we should also add a note to the common includes for plugin docs: https://github.com/andsel/logstash/pull/1